src/robots/Robot.java

```diff
@@ -0,0 +1,39 @@
+package robots;
+
+public class Robot implements Shoot, Move, Yell, Defence {
```

Comment:
your Robor class looks really like abstract class. it just contain 'common' logic/fields. make it abstract. but also read about the exact definition of 'abstract class' and add corresponding method(s).

Reply:
made class abstract, modified some methods to abstract (removed the bodies from there)